jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3579,6 +3631,11 @@ public Method getEnclosingMethod() throws SecurityException {
  * @see #isAnonymousClass()
  */
 public String getSimpleName() {
+	MetadataCache cache = getMetadataCache();
+	if (cache.cachedSimpleName != null && cache.cachedSimpleName.get() != null) {
+		return cache.cachedSimpleName.get();
```

Comment:
```suggestion
	MetadataCache cache = getMetadataCache();
	if (cache.cachedSimpleName != null) {
		/* Get a strong ref to the cached name so it can't be gc'd out from under us */	
		String cachedSimpleName = cache.cachedSimpleName.get();
		if (cachedSimpleName != null) {
			return cache.cachedSimpleName.get();
		}
```

Reply:
For strict correctness, we should probably also get a local reference to cache.cachedSimpleName so we're dealing with consistent data.

Reply:
Maybe in this case, instead of `return cache.cachedSimpleName.get();` we need to `return cachedSimpleName`? Otherwise there is still a chance that `cache.cachedSimpleName.get()` can be cleaned before return.

Reply:
Once we have a strong reference to the `referent`, the SoftReference won't be cleared.